### PR TITLE
Camera limit LAB-1429

### PR DESCRIPTION
**Pull request by katSchmid**
Adding a bounding sphere as a camera

### CODE REVIEW

**Comment by katSchmid** [comment]
image

File: Source/Scene/Camera.js [hunk 1]
```diff
@@ -29,6 +29,30 @@ import CameraFlightPath from "./CameraFlightPath.js";
 import MapMode2D from "./MapMode2D.js";
 import SceneMode from "./SceneMode.js";
 
+//PROPELLER HACK
+function clampSphere(sphere, position, result) {
+  if (
+    !defined(sphere) ||
+    !defined(position) ||
+    !position ||
+    position === Cartesian3.ZERO
```
Comment (chris-cooper) [inline Source/Scene/Camera.js]: Is this what you want?  Triple equals is like comparing pointers.  It literally has to be `Cartesian3.ZERO` not a `new Cartesian3(0,0,0)`

image

If you want to compare the value you can use `Cartesian3.equals(position, Cartesian3.ZERO)`

File: Source/Scene/Camera.js [hunk 2]
```diff
@@ -29,6 +29,30 @@ import CameraFlightPath from "./CameraFlightPath.js";
...
+  ) {
+    return position;
+  }
+
+  result = Cartesian3.clone(position, result);
+  if (Cartesian3.distance(result, sphere.center) <= sphere.radius) {
+    return result;
+  }
+  var dirV = new Cartesian3();
```
Comment (chris-cooper) [inline Source/Scene/Camera.js]: Happy to have verbose names to help others reading the code.  e.g. `directionVector = ...`

File: index.cjs [hunk 3]
```diff
@@ -4,7 +4,12 @@
 var path = require("path");
 
 // If in 'production' mode, use the combined/minified/optimized version of Cesium
-if (process.env.NODE_ENV === "production") {
+// PROPELLER HACK: Use minified code in all deployed lambda environments for compactness
```
Comment (chris-cooper) [inline index.cjs]: This will probably affect visualiser releases too right?  Do we want to retain unminified code there?

**Review by chris-cooper — APPROVED** [review]
Looks good.  Just a couple of small things.

File: Source/Scene/Camera.js (hunk 1 above)
Comment (katSchmid) [reply Source/Scene/Camera.js]: Oh right

File: index.cjs (hunk 3 above)
Comment (katSchmid) [reply index.cjs]: I am undoing this, unintentional

Comment (katSchmid) [reply index.cjs]: Restored from master https://raw.githubusercontent.com/PropellerAero/cesium/master/index.cjs oddly still shows additions

**Pull request merged**